packages/trader/src/AppV2/Components/AccumulatorStats/accumulator-stats.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { ActionSheet, Text } from '@deriv-com/quill-ui';
+import { LabelPairedChevronDownSmBoldIcon } from '@deriv/quill-icons';
+import { Localize } from '@deriv/translations';
+import { observer } from '@deriv/stores';
+import { useTraderStore } from 'Stores/useTraderStores';
+import StatsRow from './accumulator-stats-row';
+
+const AccumulatorStats = observer(() => {
+    const { ticks_history_stats = {} } = useTraderStore();
+    const [is_open, setIsOpen] = useState(false);
+    const [is_open_description, setIsOpenDescription] = useState(false);
+
+    const ticks_history = useMemo(() => {
+        return ticks_history_stats?.ticks_stayed_in ?? [];
+    }, [ticks_history_stats]);
+
+    const [animationClass, setAnimationClass] = useState('');
+    const [lastValue, setLastValue] = useState<number | null>(null);
+    const [isMovingTransition, setIsMovingTransition] = useState(false);
+
+    const rows: number[][] = useMemo(() => {
+        const row_size = 5;
+        return ticks_history.reduce<number[][]>((acc, _el, index) => {
+            if (index % row_size === 0) {
+                acc.push(ticks_history.slice(index, index + row_size));
+            }
+            return acc;
+        }, []);
+    }, [ticks_history]);
+
+    const onActionSheetClose = () => {
+        setIsOpen(false);
+        setIsOpenDescription(false);
+    };
+
+    useEffect(() => {
+        if (rows[0] && rows[0].length > 0) {
+            setAnimationClass('');
+            if (lastValue && lastValue === rows[0][1]) {
+                setTimeout(() => {
```

Comment:
It is not working without setting timeout? Sorry for silly question)
Because setState is async itself

Reply:
yes, `setState` it async, executed last. That's the reason to use `setTimeout` i guess.
Because without setTimeout, `setAnimationClass('')` above will be ignored since only the last value set into local the state variable is reflected.
@ahmadtaimoor-deriv Is that right?
Will animation break if you don't clear the classname with `setAnimationClass('')` above?

Reply:
yup that's correct . 
Yeah animation won't work if I don't clear setAnimationClass to '' on top coz once it is selected it won't trigger new animation unless it is change to "animate-error" . but we want it to show animation on each "animate-success" as well.